pubsub.go

```diff
@@ -126,38 +155,86 @@ func (p *PubsubValueStore) Subscribe(key string) error {
 	// record hasn't expired.
 	//
 	// Also, make sure to do this *before* subscribing.
-	p.ps.RegisterTopicValidator(topic, func(ctx context.Context, _ peer.ID, msg *pubsub.Message) bool {
-		return p.isBetter(key, msg.GetData())
+	myID := p.host.ID()
+	_ = p.ps.RegisterTopicValidator(topic, func(ctx context.Context, src peer.ID, msg *pubsub.Message) bool {
+		cmp := p.compare(key, msg.GetData())
+
+		return cmp > 0 || cmp == 0 && src == myID
 	})
 
 	sub, err := p.ps.Subscribe(topic)
 	if err != nil {
-		p.mx.Unlock()
 		return err
 	}
 
 	p.mx.Lock()
-	existingSub, bootstraped := p.subs[key]
+	existingSub, bootstrapped := p.subs[key]
 	if existingSub != nil {
 		p.mx.Unlock()
 		sub.Cancel()
 		return nil
 	}
 
-	p.subs[key] = sub
 	ctx, cancel := context.WithCancel(p.ctx)
-	go p.handleSubscription(sub, key, cancel)
+	p.cancels[key] = cancel
+
+	p.subs[key] = sub
+	go p.handleSubscription(ctx, sub, key, cancel)
 	p.mx.Unlock()
 
 	log.Debugf("PubsubResolve: subscribed to %s", key)
 
-	if !bootstraped {
+	if !bootstrapped {
 		// TODO: Deal with publish then resolve case? Cancel behaviour changes.
 		go bootstrapPubsub(ctx, p.cr, p.host, topic)
 	}
+
 	return nil
 }
 
+func (p *PubsubValueStore) rebroadcast(ctx context.Context) {
+	time.Sleep(p.rebroadcastInitialDelay)
```

Comment:
nit for later: this should use a select and a `time.After`.

Reply:
Done, although if we really wanted we might be able to up efficiency slightly by reusing a Timer instead of a ticker.